### ISSUE TITLE
withPageAuthRequired not redirecting to the correct url if there are multiple url params

### ISSUE THREAD

**Issue by kpatel27** [auth0/nextjs-auth0#351]
### Describe the problem
I have an SSR page that is protected by `withPageAuthRequired`. The url for this page is in the format of `http://localhost:3000/events?city=some-city&start=xxxx-xx-xx&end=xxxx-xx-xx`. 

When I'm logged in I have no problem accessing this page. When I'm not logged in I am redirected to the login page as expected. The problem is after logging in, I am redirected to `http://localhost:3000/events?city=some-city`. The rest of the url is not included in the redirect. 

### What was the expected behavior?
I should be redirected to the correct url which includes all the url/query string params. 

I checked the network tab on Chrome and this is the url requested when I'm being redirected to the login page: `http://localhost:3000/api/auth/login?returnTo=/events?city=some-city&start=xxxx-xx-xx&end=xxxx-xx-xx`

I expect the `returnTo` url param to equal `/events?city=some-city&start=xxxx-xx-xx&end=xxxx-xx-xx` but that is not the case. There are actually 3 url params here:

- `returnTo: /events?city=some-city`
- `start: xxxx-xx-xx`
- `end: xxxx-xx-xx`

The `returnTo` value will only include the url up to the first url/query string param.

### Environment

- **Version of this library used:** 1.3.0
- **Which framework are you using, if applicable:**next 10.0.7

**Comment by adamjmcgrath** [comment]
Thanks for reporting this @kpatel27 

Looks like we need to `encodeURIComponent` somewhere - will investigate

**Comment by adamjmcgrath** [comment]
@kpatel27 - have raised some work internally to fix this

**Comment by davidwterry** [comment]
Was this actually fixed?  We're using v1.9.1 and when hitting a URL that has two query parameters, only the first one is preserved after a login, which is exactly what was described in the original post.

**Comment by davidwterry** [comment]
This is what I was experiencing (I'll explain "was" in a moment).  If an app with deep links (or if I just used a bookmark) sent me to this URL *before* logging into the system:

https://cdol-beta-stage.churchofjesuschrist.org/leader/86f45503-0a8e-4558-8bf7-49f87942fd4a?cmisId=5544897553&position=702af19e-0a3a-409b-aca0-93be2bec2cc3

After logging in, the URL would be:

https://cdol-beta-stage.churchofjesuschrist.org/leader/86f45503-0a8e-4558-8bf7-49f87942fd4a?cmisId=5544897553

Sound familiar?  

About the "was"... I fixed it by getting rid of the query parameters and using a path instead.  The deep links now look like this:

https://cdol-beta-stage.churchofjesuschrist.org/leader/b3076b26-ad0c-4658-ad1a-21890c2cdce4/20825430055/c7a13517-688f-4054-bf67-c19adcb6677e

And this works just fine across a login intercept.   So to me, whatever it is doing, sure looks like a regression from the bug fix or something.  I did note that I am using version 1.9.1 (which I believe is the latest release).  So unless there is also something I have to do in my code, it doesn't feel fixed.
